tooling/nargo_cli/src/cli/debug_cmd.rs

```diff
@@ -0,0 +1,121 @@
+use acvm::acir::native_types::WitnessMap;
+use clap::Args;
+
+use nargo::artifacts::debug::DebugArtifact;
+use nargo::constants::PROVER_INPUT_FILE;
+use nargo::package::Package;
+use nargo_toml::{get_package_manifest, resolve_workspace_from_toml, PackageSelection};
+use noirc_abi::input_parser::{Format, InputValue};
+use noirc_abi::InputMap;
+use noirc_driver::{CompileOptions, CompiledProgram};
+use noirc_frontend::graph::CrateName;
+
+use noir_debugger;
+
+use super::compile_cmd::compile_bin_package;
+use super::fs::{inputs::read_inputs_from_file, witness::save_witness_to_dir};
+use super::NargoConfig;
+use crate::backends::Backend;
+use crate::errors::CliError;
+
+/// Executes a circuit in debug mode
+#[derive(Debug, Clone, Args)]
+pub(crate) struct DebugCommand {
+    /// Write the execution witness to named file
+    witness_name: Option<String>,
+
+    /// The name of the toml file which contains the inputs for the prover
+    #[clap(long, short, default_value = PROVER_INPUT_FILE)]
+    prover_name: String,
+
+    /// Detail all packages in the workspace
+    #[clap(long, conflicts_with = "package")]
+    workspace: bool,
+
+    /// The name of the package to execute
+    #[clap(long)]
+    package: Option<CrateName>,
+
+    #[clap(flatten)]
+    compile_options: CompileOptions,
+}
+
+pub(crate) fn run(
+    backend: &Backend,
+    args: DebugCommand,
+    config: NargoConfig,
+) -> Result<(), CliError> {
+    let toml_path = get_package_manifest(&config.program_dir)?;
+    let selection = args.package.map_or(PackageSelection::DefaultOrAll, PackageSelection::Selected);
+    let workspace = resolve_workspace_from_toml(&toml_path, selection)?;
+    let target_dir = &workspace.target_directory_path();
+
+    let (np_language, opcode_support) = backend.get_backend_info()?;
+    for package in &workspace {
```

Comment:
I've been playing a bit with this, and:

1. If the user provided a `--package` arg, the workspace iterator already filters out every other package.
2. If there's no `--package` provided, we can start the debugger on the first binary package visited by this loop, and break the loop after. This would have the nice added benefit of doing the expected thing in single package workspaces (that is, default to debugging the only available package).